s4e-backend/src/main/java/pl/cyfronet/s4e/sync/step/VerifyAllArtifactsExist.java

```diff
@@ -0,0 +1,67 @@
+package pl.cyfronet.s4e.sync.step;
+
+import lombok.Builder;
+import lombok.val;
+import pl.cyfronet.s4e.ex.S3ClientException;
+import pl.cyfronet.s4e.service.SceneStorage;
+import pl.cyfronet.s4e.sync.Error;
+import pl.cyfronet.s4e.sync.context.Context;
+
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static pl.cyfronet.s4e.sync.Error.*;
+
+@Builder
+public class VerifyAllArtifactsExist implements Step<Context, Error> {
+    public static final String SCENE_ARTIFACTS_PROPERTY = "artifacts";
+
+    private final Supplier<SceneStorage> sceneStorage;
+
+    private final Function<Context, JsonObject> sceneJson;
+    private final BiConsumer<Context, Map<String, String>> update;
+
+    @Override
+    public Error apply(Context context) {
+        val error = context.getError();
+        SceneStorage sceneStorage = this.sceneStorage.get();
+
+        JsonObject sceneJson = this.sceneJson.apply(context);
+
+        val artifacts = new HashMap<String, String>();
+
+        JsonObject artifactsObject = sceneJson.getJsonObject(SCENE_ARTIFACTS_PROPERTY);
+        Map<String, String> notFound = new LinkedHashMap<>();
+        for (val entry : artifactsObject.entrySet()) {
+            String name = entry.getKey();
+            String path = ((JsonString) entry.getValue()).getString();
+            if (!path.startsWith("/")) {
```

Comment:
Instead 

            if (!path.startsWith("/")) {
                return error.code(ERR_ARTIFACT_PATH_INCORRECT)
                        .parameter("artifact_" + name, path).build();
            }

there can be method
throwErrorOnInvalid(path, ...)

Whole for loop could be splited to more minimalist methods

Reply:
I'll see about this, and the following ones

Reply:
ok